### PR TITLE
Add Support for Dynamic Hidden Layers in MLP and Integration with skorch NeuralNetClassifier

### PR DESCRIPTION
**Pull request by Vendron**
# Summary
This PR introduces significant enhancements to the current MLP model by allowing dynamic configuration of multiple hidden layers and integrates the model with skorch's NeuralNetClassifier for seamless training and evaluation. This change enables the model to handle temporal vectors by flattening them and ensures more flexibility and scalability in model design.

# Changes
## Model Implementation
- The MLP model now supports an arbitrary number of hidden layers. The sizes of these hidden layers can be specified through a list.
- MLP model has been wrapped using skorch's NeuralNetClassifier, simplifying the training and evaluation process.
- Added functionality to preprocess temporal vectors by flattening them into a single vector per sample. This approach enables the MLP to process temporal data effectively.

# Testing & Evaluation
Results of this implementation have given the following results;
Accuracy: 0.926056338028169
Precision: 0.9273356401384083
Recall: 0.8948247078464107
F1 Score: 0.9107901444350043
Confusion Matrix:
[[779  42]
 [ 63 536]]
ROC-AUC Score: 0.9218337911948253

### CODE REVIEW

File: src/models/mlp.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+from typing import List
+from torch.nn import Module, Linear, ReLU, Dropout, Sigmoid, Sequential, BCELoss
+import torch.optim as optim
+from skorch import NeuralNetClassifier
+
+class MLPModule(Module):
```
Comment (simonprovost) [inline src/models/mlp.py]: Later on, nothing is critical no worries. Name it so that we know it's a Longitudinal variant. If it appears to be relevant to your MSc, Alex will undoubtedly provide you with a name; however, as of today, it is unknown. As a result, simply use LongitdinalMLP or whatever you deem more appropriate, but only to distinguish it's not pure MLP. Though nothing is critical.

File: src/models/mlp.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+        Args:
+            X (torch.Tensor): Input tensor.
+
+        Returns:
+            torch.Tensor: Output tensor.
+        """
+        return self.model(X)
+
+class MLP:
```
Comment (simonprovost) [inline src/models/mlp.py]: Similar to as previously.

File: src/models/mlp.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    Wrapper class for skorch NeuralNetClassifier with MLPModule.
```
Comment (simonprovost) [inline src/models/mlp.py]: Just curious. Cannot you inherit from NeuralNetClassifier in `class MLP` ? So that `class MLP(NeuralNetClassifier)` is actually really a wrapper. To try it out.

File: src/models/mlp.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    Attributes:
+        net (NeuralNetClassifier): skorch NeuralNetClassifier instance.
+    """
+    def __init__(self, input_size: int, hidden_sizes: List[int], output_size: int, dropout_rate: float, max_epochs: int, lr: float):
```
Comment (simonprovost) [inline src/models/mlp.py]: Good typing! I like it.

**Review by simonprovost — COMMENTED** [review]
Just some tiny comments!

File: src/models/mlp.py (hunk 1 above)
Comment (Vendron) [reply src/models/mlp.py]: Understood, I'll just rename to TemporalMLP and can work from there (so it's clearly labelled)
